### PR TITLE
feat: allow user to hide his online status

### LINKED ISSUE #766: Connect Invisible System. (Always show offline)
[//]: <Don't delete any header! Please fill out the form or your bug report will not be considered!> 

## About the new API

- Provider: `Kahla`

## About the API feature

This API produces allows the client-side to set or get the current user's invisible status.

## API document address

The document of getting status is [here](https://wiki.aiursoft.com/Kahla/Auth.md#Me)
The document of setting the status is [here](https://wiki.aiursoft.com/Kahla/Auth.md#UpdateClientSetting)

The default value is `false`. Once you enable this, you are always displayed as `offline`

## API source code

The source code of the API is [here](https://github.com/AiursoftWeb/Kahla/blob/dev/Kahla.Server/Controllers/AuthController.cs#L144)

## Menu logic

Add a new part to Kahla's advanced settings: `Privacy Settings`

In the `Privacy Settings` part, only two checkboxes:

* Hide Email
* Always offline

Remove the `Hide Email` checkbox in the profile settings.

### PR DESCRIPTION
**Pull request by hv0905**
close #766

### CODE REVIEW

**Review by Anduin2017 — APPROVED** [review]

File: tslint.json [hunk 1]
```diff
@@ -16,11 +16,6 @@
     },
     "eofline": true,
     "forin": true,
-    "import-blacklist": [
```
Comment (Anduin2017) [inline tslint.json:19]: Is this the best practice?

**Review by Anduin2017 — APPROVED** [review]

**Comment by Anduin2017** [comment]
Did you remove the email settings in the profile tab?

**Comment by hv0905** [comment]
Oh, pls wait

**Comment by hv0905** [comment]
What about change `HideMyEmail` selection from `/Auth/UpdateInfo` to UpdateClientSettings? It might be better fix the issue and make the api more reasonable. @Anduin2017

**Pull request merged**